packages/notebook-extension/src/cell-executor.ts

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import type { JupyterFrontEndPlugin } from '@jupyterlab/application';
+import { ICellExecutor } from '@jupyterlab/notebook';
+import { runCell } from '@jupyterlab/notebook';
+
+/**
+ * Notebook cell executor plugin.
+ */
+export const cellExecutor: JupyterFrontEndPlugin<ICellExecutor> = {
+  id: '@jupyterlab/notebook-extension:cell-executor',
+  description: 'Provides the notebook cell executor.',
```

Comment:
What about console cells? These get executed as well and 90% of the code is shared with notebook. Since this is named `ICellExecutor`, should we make it handle either notebook _or_ console (and then if in future another execution context comes along we can easily add it).

Reply:
I thought about it. But the two are quite different (presumably because they are separated). So should I try to reconciliate them or should the cell executor has two methods: `runNotebookCell` and `runConsoleCell`?

For reference, the console code is there:
https://github.com/jupyterlab/jupyterlab/blob/946f18cc1178155a3b979adf058918ab63fb9a60/packages/console/src/widget.ts#L724

Reply:
>  So should I try to reconciliate them or should the cell executor has two methods: `runNotebookCell` and `runConsoleCell`?

I am not sure which would be better; I brought it up on the lab call yesterday but there were no strong opinions. 

My thoughts:
- The disadvantage of reconciliating them into one method is that implementers might need to do a few conditional logic blocks depending on the execution context
- The disadvantage of two methods is that implementers would need to provide both, or we would need to have a default implementation as a fallback if we decided to make them optional (but then an interface with all methods optional risks that the implementer will fail to provide any if they make a typo and will be confused why nothing is working).

Thinking more about it, maybe there should be two separate plugins:
- `@jupyterlab/notebook-extension:cell-executor` providing `INotebookCellExecutor`
- `@jupyterlab/console-extension:cell-executor` providing `IConsoleCellExecutor`

Reply:
Separate plugins sounds better in the sense that additional execution context means just a new plugin, and that there is no need to have any fallbacks/optional methods, and no conditional logic.

The downside is that adding new execution context does require implementers to add a new plugin, but realistically this is something which does not happen frequently so I think this is a good trade-off.

Reply:
Having two plugins seems a good path. Thanks for bringing this PR for discussion at the call.

Reply:
So I think the minimum required change for this PR would be renaming `ICellExecutor` to `INotebookCellExecutor`.